signac/contrib/job.py

```diff
@@ -88,29 +229,27 @@ class Job:
     def __init__(self, project, statepoint=None, _id=None):
         self._project = project
 
+        # Prepare wd in advance so that the attribute exists in checks below.
+        self._wd = None
```

Comment:
See above comment about moving all this logic for initializing `_wd`, `_document`, etc. into a `_prepare` method.